src/main/java/org/prebid/server/bidder/valueimpression/ValueImpressionBidder.java

```diff
@@ -0,0 +1,158 @@
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.valueimpression.ExtImpValueImpression;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+import org.springframework.util.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class ValueImpressionBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpValueImpression>> VALUE_IMPRESSION_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpValueImpression>>() {
+            };
+
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public ValueImpressionBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        try {
+            final BidRequest bidRequest = processRequest(request);
+
+            return Result.of(
+                    Collections.singletonList(HttpRequest.<BidRequest>builder()
+                            .method(HttpMethod.POST)
+                            .uri(endpointUrl)
+                            .headers(HttpUtil.headers())
+                            .body(mapper.encode(bidRequest))
+                            .payload(request)
+                            .build()),
+                    Collections.emptyList()
+            );
+        } catch (PreBidException e) {
+            return Result.emptyWithError(BidderError.badInput(e.getMessage()));
+        }
+    }
+
+    private BidRequest processRequest(BidRequest bidRequest) {
+        if (CollectionUtils.isEmpty(bidRequest.getImp())) {
+            throw new PreBidException("No valid impressions in the bid request");
+        }
+        final List<Imp> validImpList = new ArrayList<>();
+        for (final Imp imp : bidRequest.getImp()) {
+            ExtImpValueImpression extImpValueImpression = parseImp(imp);
+            validImpList.add(imp.toBuilder().ext(mapper.mapper().valueToTree(extImpValueImpression)).build());
+        }
+        return bidRequest.toBuilder().imp(validImpList).build();
+    }
+
+    private ExtImpValueImpression parseImp(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), VALUE_IMPRESSION_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
```

Comment:
You just rewrite ext in imp. 
Doing this, we can exclude all fields from ext that is not in the model. But I think we can simplify this just validating that ext can be parsed. 
Or we can keep this logic. Mb they really doing this on purpose. But still, make use of Stream API instead for creating ArrayList